src/spu/src/services/public/stream_fetch.rs

```diff
@@ -312,103 +302,100 @@ impl StreamFetchHandler {
 
         let mut next_offset = offset.isolation(&self.isolation);
 
-        if file_partition_response.records.len() > 0 {
-            if let Some(module) = module_option {
-                type DefaultPartitionResponse = FetchablePartitionResponse<RecordSet>;
-
-                debug!("creating smart filter");
-                let filter_batch = {
-                    let filter = module.create_filter().map_err(|err| {
-                        IoError::new(ErrorKind::Other, format!("creating filter {}", err))
-                    })?;
-
-                    let records = &file_partition_response.records;
-
-                    filter
-                        .filter(records.raw_slice(), self.max_bytes as usize)
-                        .map_err(|err| {
-                            IoError::new(ErrorKind::Other, format!("filter err {}", err))
-                        })?
-                };
+        if file_partition_response.records.len() == 0 {
+            debug!("empty records, skipping");
+            return Ok((offset.isolation(&self.isolation), false));
+        }
```

Comment:
Reducing right-drift again. This is the contents of what used to be this `else` block: https://github.com/infinyon/fluvio/pull/1188/files#diff-5bcc5ec4673d6a5c8633195a7360c3402f873769c1faa90cf627b242179d2241L409

Reply:
Hmm. again, I suggest leave alone unless you have unit test to cover all path